analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/HardcodedIpAddressBase.cs

```diff
@@ -100,6 +115,21 @@ private void CheckForHardcodedIpAddressesInStringInterpolation(SyntaxNodeAnalysi
             && !IPAddress.IsLoopback(ipAddress)
             && !ipAddress.GetAddressBytes().All(x => x == 0); // Nonroutable 0.0.0.0 or 0::0
 
+        private bool IsInDocumentationBlock(IPAddress address) =>
+            address.AddressFamily switch
+            {
+                AddressFamily.InterNetwork =>
+                    address.GetAddressBytes() is var ip
+                    && interNetworkDocumentationRanges.Any(range => SequenceStartsWith(ip, range)),
+                AddressFamily.InterNetworkV6 =>
+                    address.GetAddressBytes() is var ip
+                    && SequenceStartsWith(ip, interNetwork6DocumentationRange),
+                _ => false,
```

Comment:
This  case is not coverable, because:
```cs
        public AddressFamily AddressFamily
        {
            get
            {
                return IsIPv4 ? AddressFamily.InterNetwork : AddressFamily.InterNetworkV6;
            }
        }
```
https://github.com/dotnet/runtime/blob/324bd24ba1b62493d6a50c13789da943920b4436/src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs#L342-L348